### ISSUE TITLE
wayland support

### ISSUE THREAD

**Issue by bijang** [leftwm/leftwm#71]
Any plans for wayland support?

**Comment by lex148** [comment]
Yes and no. When laying out the framework for Leftwm I always intended it to be able to support different backend display servers. In code their is a whole interface / trait to accomplish this. Eventually I will support Wayland and I do plan on doing this sometime down the road, but this is not a priority for me right now. At least in my mind I would like to lock down the feature set and get it as stable as a rock before attempting this. Additionally I want to do more with theming before I get to Wayland. At some point there will be a big code refactor too.

All that being said, these are my priorities. If you or anyone else wants Wayland and wants it now then by all means go for it. I LOVE pull requests and would be happy to help anyone interested in working on it.

**Comment by bijang** [comment]
Thank you for the response! 
thats the best answer to be honest. i will be following your progress! 
one last question, the theming is not in a particular language, will the the windowing config also be lang agnostic (like bspwm for example)?

**Comment by lex148** [comment]
yes

**Comment by izhakjakov** [comment]
Here is why (IMHO) wayland should be your priority: https://www.youtube.com/watch?v=8E0SOWo-Gsg
Prioritizing wayland might actually make leftwm development much easier.

**Comment by stevensonmt** [comment]
Am I correct that leftwm is architected to allow implementation of other backends via implementing the `display_server` traits for that backend? Are there other places where X11 specific code lives?

**Comment by VuiMuich** [comment]
> Am I correct that leftwm is architected to allow implementation of other backends via implementing the `display_server` traits for that backend? Are there other places where X11 specific code lives?

Keybinds rely on the `x11-dl` crate so I guess, these would need an abstraction to support a different library as well right?

**Comment by NicTanghe** [comment]
I saw someone mentioning trending towards Mir. So I guess that means Wayland support is on the radar?

**Comment by uncomfyhalomacro** [comment]
you can probably check out https://github.com/mahkoh/jay for a Wayland compositor implementation in Rust

**Comment by Mawfyy** [comment]
Hey, i wanna know how this feature goes on, some news or something?

**Comment by nemalex** [comment]
Look at https://github.com/leftwm/leftwm/issues/1035

**Comment by Mawfyy** [comment]
Thanks
